### PR TITLE
[nova] run nova-manage placement sync_aggregates in online-migration

### PR DESCRIPTION
**Pull request by mariusleu**

### CODE REVIEW

**Review by joker-official — APPROVED** [review]

**Review by joker-official — CHANGES_REQUESTED** [review]
This should only run when we're on rocky. So we either need to get the version somehow (currently no idea - maybe some secrets value or from an image?) or we need to check for the availability of the command.

File: openstack/nova/templates/bin/_db-online-migrate.tpl [hunk 1]
```diff
@@ -1,7 +1,13 @@
-#!/bin/sh
+#!/usr/bin/env bash
 
 set -x
 set -e
 
-nova-manage --config-file /etc/nova/nova.conf db online_data_migrations
+nova_manage="nova-manage --config-file /etc/nova/nova.conf"
+available_commands_text=$(nova-manage --help | awk '/Command categories/ {getline; print $0}')
```
Comment (joker-official) [inline openstack/nova/templates/bin/_db-online-migrate.tpl:7]: Does `getline` read the next line?

File: openstack/nova/templates/bin/_db-online-migrate.tpl [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 
+$nova_manage db online_data_migrations
+
+if [[ $available_commands_text == *"placement"*  ]]; then
```
Comment (joker-official) [inline openstack/nova/templates/bin/_db-online-migrate.tpl:11]: I don't feel like that's very accurate. Can we do something like
```
if echo "${available_commands_text}" | grep -q -E '[{,]placement[},]'; then
```
```
$ text='{version,bash-completion,network,cell_v2,db,cell,floating,api_db}'
$ echo "${text}" | grep -q -E '[{,]placement[},]' && echo "success"
$ text='{version,bash-completion,network,cell_v2,db,cell,floating,api_db,placement}'
$ echo "${text}" | grep -q -E '[{,]placement[},]' && echo "success"
success
```

Then we can also switch back to using `#!/bin/sh`.

File: openstack/nova/templates/bin/_db-online-migrate.tpl (hunk 1 above)
Comment (mariusleu) [reply openstack/nova/templates/bin/_db-online-migrate.tpl:7]: Yes.

**Review by joker-official — APPROVED** [review]

**Pull request merged**